File: tests/fixtures/patterns-positive/singleton/mutB.py
class AppConfig:
    _instance = None

    def __init__(self):
        self.loaded_at = None

    @staticmethod
    def get_instance():
        if AppConfig._instance is None:
            AppConfig._instance = AppConfig()
        return AppConfig._instance


# POSITIVO (mutación B: fuga de construcción). get_instance() sigue ahí, pero
# estos dos sitios construyen AppConfig directamente en vez de pasar por él.
def boot_startup():
    return AppConfig()


def boot_worker():
    return AppConfig()

File: tests/fixtures/patterns-positive/singleton/mutB.ts
class AppConfig {
  static instance: AppConfig | null = null;
  loadedAt: number;
  constructor() {
    this.loadedAt = Date.now();
  }
  static getInstance(): AppConfig {
    AppConfig.instance ||= new AppConfig();
    return AppConfig.instance;
  }
}

// POSITIVO (mutación B: fuga de construcción). getInstance() sigue ahí, pero
// estos dos sitios construyen AppConfig directamente en vez de pasar por él.
function bootStartup(): AppConfig {
  return new AppConfig();
}
function bootWorker(): AppConfig {
  return new AppConfig();
}
